app/views/devise/menu/_login_items.html.haml

```diff
@@ -16,8 +16,12 @@
     %ul.menu
       %li
         = link_to 'Edit Your Profile', profile_path
-      %li
-        = link_to 'Log Out', destroy_user_session_path, method: :delete
+      - if current_user.type === 'RegisteredUser'
```

Comment:
you can also user method  saml_user? from User class. so string check can be avoided.

Reply:
Thanks. I added those methods after This code was writing. Good catch.